woocommerce/woocommerce-ios
# Login enhancement: Add analytics for the Jetpack setup flow

Part of: #7269 


### Description

This PR adds analytics to the Jetpack setup flow:

<p></p>



<figure class="wp-block-table">

Event | Trigger | Properties
-- | -- | --
login_jetpack_setup_button_tapped | The user taps the Install Jetpack button on the Jetpack error screen. |  
login_jetpack_setup_dismissed | The user taps the Cancel button or swipes down to dismiss the modal. | source: web \| native
login_jetpack_setup_completed | The Jetpack install flow completes. | source: web \| native
login_jetpack_setup_flow | User reaches a new step in the Jetpack setup flow | source: web \| nativestep:  automatic_install \| wpcom_login \| authorize \| site_login \|  plugin_detail \| plugin_installation \| plugin_activation \| plugin_setup

Also: update the `mobile_redirect` URL with `woocommerce://jetpack-connected` since https://github.com/Automattic/wp-calypso/pull/65715 has been merged and deployed.

</figure>

### Testing instructions

1. Create a JN site with WooCommerce and without Jetpack.
2. On the app, log out if necessary, then select Enter Site Address on the prologue screen.
3. Enter the JN site address, then log in with your WP.com account.
4. Notice that the Jetpack error screen is presented because your site doesn't have Jetpack. 
5. Select the Install Jetpack button, and notice in the console: `🔵 Tracked login_jetpack_setup_button_tapped`.
6. Tap or swipe back, notice in the console: `🔵 Tracked login_jetpack_setup_dismissed` with `source: web`.
7. Select the Install Jetpack button again and follow the instructions on the web view for the setup steps. Notice for each step, there's an event in the console: `🔵 Tracked login_jetpack_setup_flow` with the correct step and `source: web`.
8. At the last step, after tapping "Authorize" for authorizing your WP.com account for Jetpack, notice in the console: `🔵 Tracked login_jetpack_setup_completed, properties: [AnyHashable("source"): "web"]`, and the app should navigate to the home screen.

---
